### PR TITLE
Closes (#1700) in docs

### PR DESCRIPTION
**Pull request by theghostmac**
All I did:
1. I made the data-collection.md file contain the **theory/definitions** of the components of OpenTel Collector. 
2. I made the configuration.md file contain **only** the configurations examples for the components and stripped all definitions. I made grammatical changes to improve technical perception.
3. I made significant changes to both files, like tables (requires adding <meta charset=”utf8”> to the static site generator to show the ticks. Or replacing the :heavy_check_mark: with something else.), and more elaborate definitions with example scripts.

### CODE REVIEW

**Comment by svrnm** [comment]
thanks @theghostmac , can you make sure that you sign the CLA & that the build works:

> 1:27:55 AM: ERROR 2022/10/05 23:27:55 docs/concepts/data-collection.md: use a local path, not an external URL, for the following reference to a local page: https://opentelemetry.io/docs/concepts/signals

File: content/en/docs/concepts/data-collection.md
Comment (tigrannajaryan) [inline content/en/docs/concepts/data-collection.md:72]: This is only a subset of the processors can do.

File: content/en/docs/concepts/data-collection.md
Comment (tigrannajaryan) [inline content/en/docs/concepts/data-collection.md:73]: Sentence unclear.

File: content/en/docs/concepts/data-collection.md
Comment (tigrannajaryan) [inline content/en/docs/concepts/data-collection.md:79]: Same here, this is not a complete list.

File: content/en/docs/concepts/data-collection.md
Comment (tigrannajaryan) [inline content/en/docs/concepts/data-collection.md:94]: What does "transport values" mean?

File: content/en/docs/concepts/what-is-opentelemetry.md
Comment (tigrannajaryan) [inline content/en/docs/concepts/what-is-opentelemetry.md:9]: Unrelated change, not part of https://github.com/open-telemetry/opentelemetry.io/issues/1700

File: content/en/docs/concepts/observability-primer.md
Comment (tigrannajaryan) [inline content/en/docs/concepts/observability-primer.md:19]: Unrelated changes, not part of https://github.com/open-telemetry/opentelemetry.io/issues/1700

File: .idea/vcs.xml
Comment (chalin) [inline .idea/vcs.xml:1]: I'd (somewhat strongly) vote against committing any `.idea` files. I'd be ok adding `/.idea` to the top-level `.gitignore` file.

**Review by chalin — CHANGES_REQUESTED** [review]
I'd (somewhat strongly) vote against committing any `.idea` files. Consider adding `/.idea` to the top-level `.gitignore` file.

WDYT @svrnm?

**Comment by svrnm** [comment]
> I'd (somewhat strongly) vote against committing any `.idea` files. Consider adding `/.idea` to the top-level `.gitignore` file.
> 
> WDYT @svrnm?

I assume this was a mistake. Let's add it to the `.gitignore`

**Comment by cartermp** [comment]
@theghostmac have you had a chance to look through feedback here?

**Comment by theghostmac** [comment]
Yes. I am trying to finish up with the work that brought me to the repository, after that, I'll fix the issues.

Thanks for reminding me.

File: content/en/docs/concepts/data-collection.md
Comment (theghostmac) [reply content/en/docs/concepts/data-collection.md:94]: To export data into multiple systems, as well as to receive/consolidate data coming from multiple systems. Does that make sense, please?

Comment (tigrannajaryan) [reply content/en/docs/concepts/data-collection.md:94]: > To export data into multiple systems, as well as to receive/consolidate data coming from multiple systems. Does that make sense, please?

Yes, this does make sense, however I do not think "transport values" explains it well. Perhaps try rewording.

**Comment by theghostmac** [comment]
@tigrannajaryan  I changed the wording to simply "export data"; would that be sufficient?

File: content/en/docs/concepts/data-collection.md
Comment (tigrannajaryan) [inline content/en/docs/concepts/data-collection.md:94]: Processor don't export data. Exporters export, processors process. I think it can be confusing to mix terminology like this.

File: content/en/docs/concepts/data-collection.md
Comment (cartermp) [reply content/en/docs/concepts/data-collection.md:72]: Yes, I think it would be better to say something like:

> The job of a processor is to process telemetry data in some way before it is exported. For example, telemetry data can redacted or filtered out based on rules specified in the collector configuration file. There are several processors available that can be configured in a processing pipeline.

File: content/en/docs/concepts/data-collection.md
Comment (cartermp) [reply content/en/docs/concepts/data-collection.md:73]: Agreed. I would recommend changing to something like:

> Processors run on data that has already been ingested by a receiver, but before that data is exported.

Or perhaps just drop it altogether since it's fairly intuitive that processors are "between" a receiver and an exporter.

File: content/en/docs/concepts/data-collection.md
Comment (cartermp) [inline content/en/docs/concepts/data-collection.md:78]: I think it would be better to not include this paragraph + table. There should arguably be an authoritative table of processors, either in the docs or in the collector repository. But I think we can tackle that orthogonally to this PR because it would involve ongoing contributions by collector contributors/maintainers to keep it up to date.

File: content/en/docs/concepts/data-collection.md
Comment (cartermp) [inline content/en/docs/concepts/data-collection.md:96]: > It is important to learn
about the types of processors in detail to maximize their usage.

Hmmmm. Is it? I mean, I think it is, but I'm not sure about this sentence. It's perfectly valid for lots of people to just stand up a "pass through" collector just so that they can manage secrets for a backend in one place. I wouldn't feel comfortable making a _best practices_ statement like this just yet, not at least without consulting several people in the community.

File: content/en/docs/concepts/data-collection.md
Comment (cartermp) [reply content/en/docs/concepts/data-collection.md:98]: Agreed, I do not think we should duplicate documentation here. This should probably be a discussion topic between Comms and Collector SIGs to figure out what we think is best for end-users, how they should be documented and given examples, and how to best manage keeping the docs up to date. I'm happy to track the outcome of that discussion as work somewhere that someone could pick up.

But for now @theghostmac if you could drop these parts from your PR that would be best. We could also put them in a temporary file (hidden by default) if they're an improvement over what's in the collector repo today.

File: content/en/docs/concepts/data-collection.md
Comment (cartermp) [inline content/en/docs/concepts/data-collection.md:182]: Similar to other comments, I think it'd probably best to not include this table here for now.

**Review by cartermp — CHANGES_REQUESTED** [review]
Thanks @theghostmac for your work so far - I am happy to see it progress and excited to get your improvements in. And thanks for being responsive to feedback on your work 🙂. I think there's still some substantive feedback to address before we can approve this, though. It's not an easy topic because there's some unresolved questions around what should/shouldn't be documented in which place and how to best maintain those docs. Thank you!

**Comment by cartermp** [comment]
Closing due to inactivity, and there's now conflicts. If you'd like to get these changes in, please ping here or resubmit and we'll take a look.

**Pull request closed without merging**
